### PR TITLE
Implement basic tests. Change project hierarchy.

### PR DESCRIPTION
**Pull request by m-kostrzewa**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"github.com/docker/go-plugins-helpers/sdk"
+)
+
+func main() {
+	d := &driver.ContrailDriver{}
+	h := network.NewHandler(d)
+
+	config := sdk.WindowsPipeConfig{
+		// This will set permissions for Everyone user allowing him to open, write, read the pipe
+		SecurityDescriptor: "S:(ML;;NW;;;LW)D:(A;;0x12019f;;;WD)",
```
Comment (Attil) [inline main.go:16]: IMO not everyone should have these permissions.

File: tests/driver_test.go [hunk 2]
```diff
@@ -42,28 +50,50 @@ var _ = Describe("Network Driver", func() {
 		})
 
 		Context("on AllocateNetwork", func() {
-			// Dunno. in standard windows drivers, it throws NotImplementedError
+			It("responds with empty AllocateNetworkResponse, nil", func() {
+				req := network.AllocateNetworkRequest{}
+				resp, err := contrail_driver.AllocateNetwork(&req)
+				Expect(resp).To(Equal(&network.AllocateNetworkResponse{}))
+				Expect(err).ShouldNot(HaveOccurred())
+			})
```
Comment (Attil) [inline tests/driver_test.go:58]: It should check whether the network was allocated.

File: tests/driver_test.go [hunk 3]
```diff
@@ -42,28 +50,50 @@ var _ = Describe("Network Driver", func() {
...
 		})
 
 		Context("on DeleteNetwork", func() {
-			It("responds with nil", func() {})
+			It("responds with nil", func() {
+				req := network.DeleteNetworkRequest{}
+				err := contrail_driver.DeleteNetwork(&req)
+				Expect(err).ShouldNot(HaveOccurred())
```
Comment (Attil) [inline tests/driver_test.go:65]: It should check whether the network was deleted.

File: tests/driver_test.go [hunk 4]
```diff
@@ -42,28 +50,50 @@ var _ = Describe("Network Driver", func() {
...
+			})
 		})
 
 		Context("on FreeNetwork", func() {
-			// Dunno. in standard windows drivers, it throws NotImplementedError
+			It("responds with nil", func() {
+				req := network.FreeNetworkRequest{}
+				err := contrail_driver.FreeNetwork(&req)
+				Expect(err).ShouldNot(HaveOccurred())
```
Comment (Attil) [inline tests/driver_test.go:73]: It should check whether the network was freed.

File: tests/driver_test.go [hunk 5]
```diff
@@ -42,28 +50,50 @@ var _ = Describe("Network Driver", func() {
...
 			It("allocates Contrail resources", func() {})
 			It("configures container network via HNS", func() {})
 			It("configures vRouter agent", func() {})
-			It("responds with proper CreateEndpointResponse", func() {})
+			It("responds with proper CreateEndpointResponse, nil", func() {
+				req := network.CreateEndpointRequest{}
+				resp, err := contrail_driver.CreateEndpoint(&req)
+				Expect(resp).To(Equal(&network.CreateEndpointResponse{}))
+				Expect(err).ShouldNot(HaveOccurred())
+			})
```
Comment (Attil) [inline tests/driver_test.go:86]: It should check whether the endpoint exists.

File: tests/driver_test.go [hunk 6]
```diff
@@ -42,28 +50,50 @@ var _ = Describe("Network Driver", func() {
...
 
 		Context("on DeleteEndpoint", func() {
 			It("deallocates Contrail resources", func() {})
 			It("configures vRouter Agent", func() {})
-			It("reponds with nil", func() {})
+			It("responds with nil", func() {
+				req := network.DeleteEndpointRequest{}
+				err := contrail_driver.DeleteEndpoint(&req)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
```
Comment (Attil) [inline tests/driver_test.go:96]: It should check whether the endpoint was deleted.

File: tests/driver_test.go [hunk 7]
```diff
@@ -78,34 +108,49 @@ var _ = Describe("Network Driver", func() {
...
 				It("responds with err", func() {})
 			})
-			// Dunno. in standard windows drivers, it's an empty func otherwise
 		})
 
 		Context("on Leave", func() {
+
 			Context("queried endpoint exists", func() {
-				It("responds with proper JoinResponse", func() {}) // nil maybe?
+				It("responds with proper JoinResponse, nil", func() {}) // nil maybe?
```
Comment (Attil) [inline tests/driver_test.go:116]: Check whether the endpoint left (the network?).

**Review by Attil — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (m-kostrzewa) [reply main.go:16]: I'm not sure what they should be, I created an issue for this so we can fix it in another PR (JW-15)

**Comment by m-kostrzewa** [comment]
TBH i think this PR can be closed since https://github.com/codilime/contrail-windows-docker/pull/6 contains it

File: tests/driver_test.go (hunk 2 above)
Comment (m-kostrzewa) [reply tests/driver_test.go:58]: All these change request are about non-existing features, they will be added in another pr.

**Pull request closed without merging**